packages/fuels-contract/src/execution_script.rs

```diff
@@ -0,0 +1,130 @@
+use anyhow::Result;
+use std::fmt::Debug;
+
+use fuel_gql_client::fuel_tx::{Receipt, Transaction};
+use fuel_gql_client::fuel_types::AssetId;
+
+use fuel_tx::{Checkable, ScriptExecutionResult};
+use fuels_core::parameters::TxParameters;
+use fuels_signers::provider::Provider;
+use fuels_signers::{Signer, WalletUnlocked};
+
+use fuels_types::errors::Error;
+
+use std::vec;
+
+use crate::contract::ContractCall;
+use crate::contract_calls_utils::{
+    build_script_data_from_contract_calls, calculate_required_asset_amounts, get_data_offset,
+    get_instructions, get_transaction_inputs_outputs,
+};
+
+/// TransactionExecution provides methods to create and a call/simulate a transaction that carries
+/// out contract method calls or script calls
+#[derive(Debug)]
+pub struct TransactionExecution {
```

Comment:
I would change `TransactionExecution` to something else. IMO it is a bit confusing. Unfortunately, I do not have a good alternative name. Maybe some other reviewer will have something...

Reply:
Yeah I'm not really a fan of this naming either…

Reply:
Maybe a trait would be more appropriate here. Something that implies that the transaction will be processed by the VM interpreter. Is `ExecutableTransaction` too vague?

Reply:
On second thought, `ExecutableTransaction` might not be ideal since fuel-vm implements a trait with the same name and we make se of it in our provider.

Reply:
Good call y'all. `TransactionExecution` had me very confused. 

`ExecutableSDKTransaction ` is a no for me.

I'm also not liking having `VM` all over the codebase now. It's not very descriptive; what VM? I'd suggest replacing `VM` with `Fuel`. In this case, `ExecutableFuelCall`. SDK users won't be very interested in lower-level concepts, such as virtual machines (that's for us, SDK developers, to take care of). 

The same applies to `VMCallResponse` I've been seeing in this PR. `FuelCallResponse` would be at a more appropriate abstraction level.

Reply:
Going for `FuelCallResponse` and `ExecutableFuelCall` then, unless more objections!

Reply:
Do you think `ExecutableFuelCall` should be a trait? If it's a trait it would only have one implementation AFAIU, so I'm not sure it's really the best idea. @digorithm @Br1ght0ne @MujkicA @hal3e

Reply:
Hmmm, I'm not 100% sold on the idea of this being a trait 🤔

Reply:
Ok so I'm not making it a trait for now then.